### PR TITLE
MQE-942: MFTF Annotation Static Check

### PR DESCRIPTION
**Pull request by tomreece**
### Description
* Add a new static check that validates test annotations
* Run via `bin/mftf static-checks annotations`

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/verification tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)
 - [ ] Changes to Framework doesn't have backward incompatible changes for tests or have related Pull Request with fixes to tests

### CODE REVIEW

File: src/Magento/FunctionalTestingFramework/Test/Util/AnnotationExtractor.php [hunk 1]
```diff
@@ -59,8 +59,9 @@ public function __construct()
      * @param string $filename
      * @return array
      * @throws XmlException
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
```
[the remark this answers is not in the document]
Comment (tomreece) [reply src/Magento/FunctionalTestingFramework/Test/Util/AnnotationExtractor.php:63]: Fixed

File: src/Magento/FunctionalTestingFramework/Test/Util/TestObjectExtractor.php [hunk 2]
```diff
@@ -87,7 +87,7 @@ public function getAnnotationExtractor()
      * @return TestObject
      * @throws \Exception
      */
-    public function extractTestData($testData)
+    public function extractTestData($testData, $validateAnnotations = true)
```
[the remark this answers is not in the document]
Comment (tomreece) [reply src/Magento/FunctionalTestingFramework/Test/Util/TestObjectExtractor.php:91]: Fixed

**Comment by tomreece** [comment]
> Also noticed, static check does not validate issueId for skipped tests cannot be empty check from AnnotationExtractor. Should it be included here?

@soumyau I was going to add it, but then I noticed that it's an XSD level check and will happen before we even make it to my code.
Is that ok?

Also I'm writing a manual test case in our regression backlog for this now.

**Comment by soumyau** [comment]
> > Also noticed, static check does not validate issueId for skipped tests cannot be empty check from AnnotationExtractor. Should it be included here?
> 
> @soumyau I was going to add it, but then I noticed that it's an XSD level check and will happen before we even make it to my code.
> Is that ok?
> 
> Also I'm writing a manual test case in our regression backlog for this now.

@tomreece I noticed that like XSD check does not throw error if `value` attribute is set to blank. I thought that's what the annotation check catches. Just double check, if not can close this.

**Comment by tomreece** [comment]
@soumyau It was easy to add so I added a check that detects if `issueId` is empty when a test is skipped. Please re-review.

**Pull request merged**